sql/src/main/java/io/crate/expression/scalar/arithmetic/ArithmeticFunctions.java

```diff
@@ -109,15 +110,20 @@ public static void register(ScalarFunctionModule module) {
             (arg0, arg1) -> arg0 / arg1,
             (arg0, arg1) -> arg0 / arg1
         ));
-        module.register(Names.MODULUS, new ArithmeticFunctionResolver(
-            Names.MODULUS,
-            "%",
-            FunctionInfo.DETERMINISTIC_ONLY,
-            (arg0, arg1) -> arg0 % arg1,
-            (arg0, arg1) -> arg0 % arg1,
-            (arg0, arg1) -> arg0 % arg1,
-            (arg0, arg1) -> arg0 % arg1
-        ));
+
+        java.util.function.Function<String, ArithmeticFunctionResolver> modFunctionResolverFactory =
+            name -> new ArithmeticFunctionResolver(
+                name,
+                "%",
+                FunctionInfo.DETERMINISTIC_ONLY,
+                (arg0, arg1) -> arg0 % arg1,
+                (arg0, arg1) -> arg0 % arg1,
+                (arg0, arg1) -> arg0 % arg1,
+                (arg0, arg1) -> arg0 % arg1
+            );
+
+        module.register(Names.MODULUS, modFunctionResolverFactory.apply(Names.MODULUS));
```

Comment:
lgtm

just a side note, this function is also registered for [interval and timestamp types](https://github.com/crate/crate/pull/9640/files#diff-d4be2122aeae9ba7a07ccf87d4e864b5R64). not sure if it makes sense.

Reply:
Yes good point, also even if not making sense it won't brake. We could remove that support later on if you think we should.